cli/popper/runner_slurm.py

```diff
@@ -1,249 +1,144 @@
-                    spawned_processes=self.output_stream_pid)
-
-    def _stream_error(self, err_file):
-        self.error_stream_pid = set()
-        pu.exec_cmd(["tail", "-f", err_file],
-                    spawned_processes=self.error_stream_pid)
+        self._spawned_jobs = set()
 
-    def start_output_error_stream(self, out_file, err_file):
-        self.output_stream_thread = threading.Thread(
-            target=self._stream_output, args=(out_file,))
+    def _tail_output(self, out_file):
+        self._out_stream_pid = set()
+        HostRunner._exec_cmd(["tail", "-f", out_file],
+                             pids=self._out_stream_pid)
 
-        self.error_stream_thread = threading.Thread(
-            target=self._stream_error, args=(err_file,))
+    def _start_out_stream(self, out_file, err_file):
+        self._out_stream_thread = threading.Thread(
+            target=self._tail_output, args=(out_file,))
+        self._out_stream_thread.start()
 
-        self.output_stream_thread.start()
-        self.error_stream_thread.start()
+    def _stop_out_stream(self):
+        out_stream_proc = list(self._out_stream_pid)[0]
+        out_stream_proc.kill()
+        self._out_stream_thread.join()
 
-    def stop_output_error_stream(self):
-        output_stream_proc = list(self.output_stream_pid)[0]
-        error_stream_proc = list(self.error_stream_pid)[0]
-
-        output_stream_proc.kill()
-        error_stream_proc.kill()
-
-        self.output_stream_thread.join()
-        self.error_stream_thread.join()
-
-    @staticmethod
-    def generate_script(cmd, job_name, job_script):
-        with open(job_script, "w") as f:
-            f.write("#!/bin/bash\n")
-            f.write(cmd)
-
-    @staticmethod
-    def touch_log_files(out_file, err_file):
-        if os.path.exists(out_file):
-            os.remove(out_file)
+    def _submit_batch_job(self, cmd, step):
+        job_name = pu.sanitized_name(step['name'], self._config.wid)
+        temp_dir = "/tmp/popper/slurm/"
+        os.makedirs(temp_dir, exist_ok=True)
 
-        if os.path.exists(err_file):
-            os.remove(err_file)
+        job_script = os.path.join(temp_dir, f'{job_name}.sh')
+        out_file = os.path.join(temp_dir, f'{job_name}.out')
 
-        sh.touch(out_file)
-        sh.touch(err_file)
+        # create/truncate log
+        with open(out_file, 'w'):
+            pass
 
-    def submit_batch_job(self, cmd, step):
-        job_name = pu.sanitized_name(step['name'], self.config.wid)
-        temp_dir = "/tmp/popper/slurm/"
-        os.makedirs(temp_dir, exist_ok=True)
+        with open(job_script, 'w') as f:
+            f.write('#!/bin/bash\n')
+            f.write('\n'.join(cmd))
 
-        job_script = os.path.join(temp_dir, f"{job_name}.sh")
-        out_file = os.path.join(temp_dir, f"{job_name}.out")
-        err_file = os.path.join(temp_dir, f"{job_name}.err")
+        sbatch_cmd = f'sbatch --wait --job-name {job_name} --output {out_file}'
```

Comment:
@ivotron Shouldn't there be a `--error {err_file}` also. So, if there is an error in the command running in the job script, that could be streamed to the user. Maybe we can have 2 threads, one looking at the out file and other looking at the err file. What do you say ?

Reply:
when no `--error` is given, sbatch combines the output of stderr and stdout in the same file.

Reply:
Oh, sorry, i didn't knew it.